ci-operator/step-registry/workers-rhel/pre-hook/sshtunnel/workers-rhel-pre-hook-sshtunnel-commands.sh

```diff
@@ -0,0 +1,190 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+set -x
+
+
+export BASTION_SSH_PRIV_KEY_PATH=${CLUSTER_PROFILE_DIR}/ssh-privatekey
+echo "RHEL VERSION: '${PLATFORM_VERSION}'"
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+declare bastion_private_address
+declare bastion_ssh_user
+if [ -f "${SHARED_DIR}"/bastion_private_address ] && [ -f "${SHARED_DIR}"/bastion_ssh_user ] ; then
+    bastion_private_address=$(< "${SHARED_DIR}"/bastion_private_address)
+    bastion_ssh_user=$(< "${SHARED_DIR}"/bastion_ssh_user)
+else
+    echo "File bastion_private_address does not exist in SHARED_DIR, no need to deploy ssh tunnel service!"
+    exit 0
```

Comment:
If user decided to insert this ref as a step in his/her own workflow, then bastion host is a MUST, so I think maybe `exit 1` should be more reasonable. 

BTW, a minor suggestion for the above message - "File bastion_private_address or  bastion_ssh_user does not exist in SHARED_DIR"

Reply:
updated